hil/config.py

```diff
@@ -89,3 +170,4 @@ def setup(filename='hil.cfg'):
     load(filename)
     configure_logging()
     load_extensions()
+    validate_config()
```

Comment:
move this before `configure_logging()` and this PR itself can take care of #976 
This [line](https://github.com/CCI-MOC/hil/blob/master/hil/config.py#L62) is probably what's causing trouble.

Reply:
We should probably check this by catching the IOError and reporting the issue; conceptually doing it here is weird, because the problem is an absent or non-writable directory, not a malformed confg file. Slightly more pragmatically, there's a race condition where if the directory is deleted/permissions changed between the check and the use, you could still get the error.

Reply:
@zenhack I agree that conceptually it's strange to do it here as well. For the scope of this PR, I'm thinking about moving the config validation up and then only checking for a malformed path.  The IO check can be done in another small PR.

Reply:
Sounds good.